cmd/ui-errors-utils.go

```diff
@@ -0,0 +1,158 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+)
+
+// uiErr is a structure which contains all information
+// to print a fatal error message in json or pretty mode
+// uiErr implements error so we can use it anywhere
+type uiErr struct {
+	msg    string
+	detail string
+	action string
+	help   string
+}
+
+// Return the error message
+func (u uiErr) Error() string {
+	if u.detail == "" {
+		return u.msg
+	}
+	return u.detail
+}
+
+// Replace the current error's message
+func (u uiErr) Msg(m string, args ...interface{}) uiErr {
+	return uiErr{
+		msg:    fmt.Sprintf(m, args...),
+		detail: u.detail,
+		action: u.action,
+		help:   u.help,
+	}
+}
+
+type uiErrFn func(err error) uiErr
+
+// Create a UI error generator, this is needed to simplify
+// the update of the detailed error message in several places
+// in Minio code
+func newUIErrFn(msg, action, help string) uiErrFn {
+	return func(err error) uiErr {
+		u := uiErr{
+			msg:    msg,
+			action: action,
+			help:   help,
+		}
+		if err != nil {
+			u.detail = err.Error()
+		}
+		return u
+	}
+}
+
+// typedErrToUIErr transforms Minio error to a generic uiErr
+func typedErrToUIErr(err typedError) uiErr {
+	switch err.code {
+	case unexpectedBackendVersionErrCode:
+		return uiErrUnexpectedBackendVersion(err)
+	case invalidAddressFlagErrCode:
+		return uiErrInvalidAddressFlag(err)
+	case invalidErasureEndpointsErrCode:
+		return uiErrInvalidErasureEndpoints(err)
+	case invalidFSEndpointErrCode:
+		return uiErrInvalidFSEndpoint(err)
+	}
+
+	return uiErrUnexpectedError(err)
+}
+
+// errorToUIError inspects the passed error and transforms it
+// to the appropriate UI error.
+func errorToUIErr(err error) uiErr {
+	// If this is already a uiErr, do nothing
+	if e, ok := err.(uiErr); ok {
+		return e
+	}
+
+	// If this is a minio typed error, find the most corresponding
+	// ui error and return it.
+	if e, ok := err.(typedError); ok {
+		return typedErrToUIErr(e)
+	}
+
+	// The remaining code shows generic UI error to some
+	// known errors which occur in many places, such as
+	// permission denied to access to a file or dir
+	switch e := err.(type) {
+	case *net.OpError:
+		if e.Op == "listen" {
+			return uiErrPortAlreadyInUse(e).Msg("Port " + e.Addr.String() + " is already in use")
+
+		}
+	case *os.PathError:
+		if os.IsPermission(e) {
+			return uiErrNoPrivilegesToAccessDirFiles(e).Msg("Missing privileges to access `" + e.Path + "` path")
+		}
+	}
+
+	switch err {
+	case io.ErrUnexpectedEOF:
+		return uiErrUnexpectedDataContent(err)
+	}
+
+	// Failed to identify what type of error this, return a simple UI error
+	return uiErr{msg: err.Error()}
+}
+
+// renderError() transforms a Fatal error message to a more understood error
+// and with some cool colors and rendering
```

Comment:
A comment should be  a bit more formal in tone, rather than saying `cool colors`